packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts

```diff
@@ -36,114 +46,80 @@ describe('trackLoadingTime', () => {
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, {
-        loadEventEnd: LOAD_EVENT_AFTER_ACTIVITY_TIMING,
-      }),
-    ])
-
+    setLoadEvent(LOAD_EVENT_AFTER_ACTIVITY_TIMING)
     domMutationObservable.notify()
     clock.tick(AFTER_PAGE_ACTIVITY_END_DELAY)
 
-    clock.tick(THROTTLE_VIEW_UPDATE_PERIOD)
-
-    expect(getViewUpdateCount()).toEqual(2)
-    expect(getViewUpdate(1).commonViewMetrics.loadingTime).toEqual(LOAD_EVENT_AFTER_ACTIVITY_TIMING)
+    expect(loadingTimeCallback).toHaveBeenCalledOnceWith(LOAD_EVENT_AFTER_ACTIVITY_TIMING)
   })
 
   it('should use computed loading time for initial view when load event is smaller than computed loading time', () => {
-    const { lifeCycle, domMutationObservable, clock } = setupBuilder.build()
-    const { getViewUpdate, getViewUpdateCount } = viewTest
-
-    expect(getViewUpdateCount()).toEqual(1)
+    loadType = ViewLoadingType.INITIAL_LOAD
+    const { domMutationObservable, clock } = setupBuilder.build()
 
     clock.tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
-    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, {
-        loadEventEnd: LOAD_EVENT_BEFORE_ACTIVITY_TIMING,
-      }),
-    ])
+
+    setLoadEvent(LOAD_EVENT_BEFORE_ACTIVITY_TIMING)
+
     domMutationObservable.notify()
     clock.tick(AFTER_PAGE_ACTIVITY_END_DELAY)
-    clock.tick(THROTTLE_VIEW_UPDATE_PERIOD)
 
-    expect(getViewUpdateCount()).toEqual(2)
-    expect(getViewUpdate(1).commonViewMetrics.loadingTime).toEqual(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+    expect(loadingTimeCallback).toHaveBeenCalledOnceWith(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
   })
 
   it('should use computed loading time from time origin for initial view', () => {
+    loadType = ViewLoadingType.INITIAL_LOAD
+
     // introduce a gap between time origin and tracking start
     // ensure that `load event > activity delay` and `load event < activity delay + clock gap`
     // to make the test fail if the clock gap is not correctly taken into account
     const CLOCK_GAP = (LOAD_EVENT_AFTER_ACTIVITY_TIMING - BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY + 1) as Duration
 
     setupBuilder.clock!.tick(CLOCK_GAP)
 
-    const { lifeCycle, domMutationObservable, clock } = setupBuilder.build()
-    const { getViewUpdate, getViewUpdateCount } = viewTest
-
-    expect(getViewUpdateCount()).toEqual(1)
+    const { domMutationObservable, clock } = setupBuilder.build()
 
     clock.tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
 
-    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, {
-        loadEventEnd: LOAD_EVENT_BEFORE_ACTIVITY_TIMING,
-      }),
-    ])
+    setLoadEvent(LOAD_EVENT_BEFORE_ACTIVITY_TIMING)
 
     domMutationObservable.notify()
     clock.tick(AFTER_PAGE_ACTIVITY_END_DELAY)
-
     clock.tick(THROTTLE_VIEW_UPDATE_PERIOD)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏this `THROTTLE_VIEW_UPDATE_PERIOD` could probably be removed

```suggestion
```